packages/file_selector/file_selector_web/lib/src/dom_helper.dart

```diff
@@ -14,50 +14,51 @@ class DomHelper {
 
   /// Default constructor, initializes the container DOM element.
   DomHelper() {
-    final body = querySelector('body');
+    final body = querySelector('body')!;
     body.children.add(_container);
   }
 
   /// Sets the <input /> attributes and waits for a file to be selected.
   Future<List<XFile>> getFiles({
     String accept = '',
     bool multiple = false,
-    @visibleForTesting FileUploadInputElement input,
+    @visibleForTesting FileUploadInputElement? input,
   }) {
-    final Completer<List<XFile>> _completer = Completer();
-    input = input ?? FileUploadInputElement();
+    final Completer<List<XFile>> completer = Completer();
+    final FileUploadInputElement inputElement =
+        input ?? FileUploadInputElement();
 
     _container.children.add(
-      input
+      inputElement
         ..accept = accept
         ..multiple = multiple,
     );
 
-    input.onChange.first.then((_) {
-      final List<XFile> files = input.files.map(_convertFileToXFile).toList();
+    inputElement.onChange.first.then((_) {
+      final List<XFile> files = input!.files!.map(_convertFileToXFile).toList();
       input.remove();
-      _completer.complete(files);
+      completer.complete(files);
     });
 
-    input.onError.first.then((event) {
-      final ErrorEvent error = event;
+    inputElement.onError.first.then((event) {
+      final ErrorEvent error = event as ErrorEvent;
       final platformException = PlatformException(
         code: error.type,
         message: error.message,
       );
-      input.remove();
-      _completer.completeError(platformException);
+      input!.remove();
```

Comment:
`inputElement`? This isn't safe otherwise.

Reply:
Ah yes, good catch! inputElement everywhere in this file, after the first initialization!